### ISSUE TITLE
Confusing license terms

### ISSUE THREAD

**Issue by svenssonaxel** [mattermost/mattermost-server#22864]
#### Problem

The licence says that I can use the source code
> Under the Free Software Foundation’s GNU AGPL v.3.0, subject to the exceptions outlined in this policy; or

but I don't understand what "this policy" refers to, neither what the restrictions are beyond what's already in the AGPL.
Does this mean I can use the code under the AGPL, or are there more restrictions in some other policy and if so, where?

#### Possible fixes
Clarify by rephrasing to one of the following

* > Under the Free Software Foundation’s GNU AGPL v.3.0; or
* > Under the Free Software Foundation’s GNU AGPL v.3.0 (subject to the exceptions outlined in GNU AGPL v.3.0); or
* > Under the Free Software Foundation’s GNU AGPL v.3.0, on the condition that you also adhere to the trademark policy below; or
* > Under the Free Software Foundation’s GNU AGPL v.3.0, provided that you also obey the policy found [here](URL); or

**Comment by svenssonaxel** [comment]
@amyblais any update?

**Comment by amyblais** [comment]
@svenssonaxel Are you able to provide some more context on why you are asking about this? Are you looking to use our source code?

**Comment by svenssonaxel** [comment]
Yes, I want to fix some bugs and self host an instance with those fixes without waiting for the long release cycle. In the future I might want to add some functionality too.

If Mattermost is open source, then I believe there should be no legal obstacle in doing that, at least (in the AGPL case) not after publishing my changes under the same license.

However, with the current wording it is not clear to me whether Mattermost is open source.

I understand that Mattermost the company has an interest in contributors signing over copyright for, not just licensing, their patches. I for one would have no problem doing so - if Mattermost is open source.

**Comment by amyblais** [comment]
@svenssonaxel Please take a look at our documentation on open source licensing questions:  https://docs.mattermost.com/about/faq-license.html for guidance on their questions. 
We will consider whether a clarification is needed to the license wording.

**Comment by svenssonaxel** [comment]
The link you provided talks about "Mattermost Team Edition (Open Source)" and says that this is licenced under MIT. If that's true then that's great, but I can't find the source code for it. I've only found https://github.com/mattermost/mattermost/. If this is not the repository for the Open Source Team Edition, then could you please provide a link to it? If that is the correct repository, then:
* It does not seem to be MIT licensed; only binaries compiled by Mattermost are "MIT" licensed, but cannot be considered Open Source since (I assume) they do not include the source code. If you do not intend to provide this source code under the MIT license, then perhaps the documentation should be updated so that it does not claim so?
* The source in the repository is not MIT licensed but rather maybe-almost-AGPL + terms in NOTICE.txt. AGPL is an Open Source license, but as stated above is "subject to the exceptions outlined in this policy", without clarifying what policy is referred to. Since I don't know what the exceptions are, I don't know what the license is and I don't know whether it's Open Source. If your intention is to license this repository under an Open Source license, it would be nice if you could do so clearly, and if not you probably shouldn't claim so in your documentation.

**Comment by svenssonaxel** [comment]
@amyblais Any update?

**Comment by svenssonaxel** [comment]
@amyblais My purpose is simply to self-host a Mattermost instance for hobby use by 5 people, and I still don't know how to do this in a compliant way.

My best attempt at understanding your terms is that I may use the source code under AGPL v3.0, as long as I publish any modifications and also remove any Mattermost logos and trademarks.

Is this a correct understanding?

**Comment by svenssonaxel** [comment]
I just spotted another source of confusion:
Your license file says "You are licensed" for the compiled versions and admin tools, but for use under the AGPL it says "You **may** be licensed". I'm not sure what determines whether I actually am licensed then.

What do I need to do in order to be licensed to self host a Mattermost instance?

What do I need to do in order to be allowed to compile altered source code and then run that software?
